### PR TITLE
Refactor Process Stats and Bugfixes

### PR DESCRIPTION
**Pull request by skrashevich**
- fix cpu usage statistics bugs introduced in #6298 (my apologies)
- refactor the handling of process statistics (introduced in #6276) to the more right way (IMHO)

### CODE REVIEW

File: frigate/util.py [hunk 1]
```diff
@@ -806,7 +806,20 @@ def get_cpu_stats() -> dict[str, dict]:
...
+
+                clk_tck = os.sysconf(os.sysconf_names["SC_CLK_TCK"])
+
+                process_utime_sec = utime // clk_tck
+                process_stime_sec = stime // clk_tck
+                process_starttime_sec = starttime // clk_tck
+
+                process_elapsed_sec = system_uptime_sec - process_starttime_sec
+                process_usage_sec = process_utime_sec + process_stime_sec
+                cpu_usage = process_usage_sec * 100 // process_elapsed_sec
```
Comment (skrashevich) [inline frigate/util.py:822]: fix for #6298

**Review by NickM-27 — APPROVED** [review]

**Comment by NickM-27** [comment]
These CPU usages coming in now are still pretty inconsistent compared to when top was being used. Some are a bit higher and others are continually showing as 0% when they were at 2% with top

**Comment by skrashevich** [comment]
Because of top shows cpu utilisation in moment, unlike this code: it’s get average utilization by process life time…
I understand that the state "in moment" are perceived more intuitively, but so far it has not been possible to implement code that would work normally on all systems. I'll try again...

**Comment by NickM-27** [comment]
> Because of top shows cpu utilisation in moment, unlike this code: it’s get average utilization by process life time… I understand that the state "in moment" are perceived more intuitively, but so far it has not been possible to implement code that would work normally on all systems. I'll try again...

I'm not sure it's necessary to redo the logic. If that is how it works, that is potentially more useful information anyway. Perhaps we just need this to be explained in the stats page

**Review by NickM-27 — APPROVED** [review]

**Comment by skrashevich** [comment]
> If that is how it works, that is potentially more useful information anyway (IMO)
issue for discussion. CPU utilisation  over a long period is something like an average hospital temperature, including a morgue.

**Comment by NickM-27** [comment]
Yeah, I imagine for things like the detect process which will see (relative) CPU spikes, lifetime average gives a better outlook on what CPU is used by that process vs having to refresh at exactly the moment lots of activity is going on v.s. the moment after when the scene is calm

**Comment by skrashevich** [comment]
the last commit makes data similar to `top` output. I don't like this implementation, but it works...
however, we can combine this (In-moment-value) and previous (average from period). What do you think about it?

**Comment by NickM-27** [comment]
I'll defer to @blakeblackshear on that

**Comment by skrashevich** [comment]
I added both values in the last commit

image

File: web/src/routes/System.jsx [hunk 2]
```diff
@@ -357,13 +359,15 @@ export default function System() {
                     <Thead>
                       <Tr>
                         <Th>CPU %</Th>
+                        <Th>AvgCPU %</Th>
```
Comment (NickM-27) [inline web/src/routes/System.jsx:362]:
```suggestion
                        <Th>Avg CPU %</Th>
```

File: web/src/routes/System.jsx [hunk 3]
```diff
@@ -347,23 +349,26 @@ export default function System() {
 
           <Heading size="lg">Other Processes</Heading>
           <div data-testid="cameras" className="grid grid-cols-1 3xl:grid-cols-3 md:grid-cols-2 gap-4">
-            {['go2rtc', 'recording'].map((process) => (
+            {processesNames.map((process) => (
               <div key={process} className="dark:bg-gray-800 shadow-md hover:shadow-lg rounded-lg transition-shadow">
                 <div className="capitalize text-lg flex justify-between p-4">
                   <div className="text-lg flex justify-between">{process}</div>
+                  <p>P-ID: {processes[process]['pid'] || '- '}</p>
```
Comment (NickM-27) [inline web/src/routes/System.jsx:356]: This should be done the same way as other cards as part of the table

Comment (skrashevich) [reply web/src/routes/System.jsx:356]: If you reference to other cards, all "other processes" should be a table inside one card in this logic...

Comment (NickM-27) [reply web/src/routes/System.jsx:356]: I don't think so. For example when you have multiple detectors

Comment (skrashevich) [reply web/src/routes/System.jsx]: make sense... (https://github.com/blakeblackshear/frigate/pull/6344/commits/4adf482475823ce583f77218401ab883d8a40653)
image

**Comment by NickM-27** [comment]
Is this ready for final review or do you have other changes coming?

**Comment by skrashevich** [comment]
> Is this ready for final review or do you have other changes coming?

Ready. Just a question about CPU utilisation counting method...

**Comment by blakeblackshear** [comment]
Let's just use this as is for a bit and we can always improve it again

**Pull request merged**